### ISSUE TITLE
[feature] Permit transcoding / reduced bitrates

### ISSUE THREAD

**Issue by attie** [leinelissen/jellyfin-audio-player#118]
It would be great if we can listen & download items in a lighter format - for example, I keep audio as FLAC... but MP3 would be fine on mobile (perhaps even preferable, for both storage & data allowances).

**Comment by leinelissen** [comment]
That should be perfectly possible as we can already transform files as we please. It won't be high on my priority list though. If you want some pointers to get started yourself, please let me know!

**Comment by attie** [comment]
Great! Thanks for considering it (and your great work so far!)
I'm not sure I'm well placed to look into this right now, but will keep it in mind.

For clarity - I believe we can request that media is transcoded by the server(?) which would be the preference here, rather than transcoding on-device.

**Comment by GitAkash** [comment]
Heya, @leinelissen 

I just started using the app and i really love the design,
I would like to add that this would be useful for me too. (In the train etc playing my FLAC audio would cost me a lot of data). If you plan to add this feature in the future, would you also be able or maybe earlier be able to make some kind of display of what kind of audiofile is playing? (FLAC, MP3, etc.. for cleanliness you could of course rename FLAC ALAC etc to just lossless). I myself can't code so its just maybe something to be added on the list.

**Comment by leinelissen** [comment]
Hey @GitAkash, that's a great suggestion. I will need to solve to transcoding playback issues first, but this seems like a solid approach. Let me see what I can do.

**Comment by leinelissen** [comment]
Let's expose it as a global setting right now. It's currently already a part of the settings in `src/store` but we're not passing it through yet. Since we're already querying the store in `generateTrack`, you can just fetch the value there and then pass it on to Jellyfin. Now the last thing you'll need to implement is some UI in `src/screens/Settings`. Let's put it in a separate page with some introduction and a limited set of options, such as these:
- 5Mbit (practically unlimited, this should be the default)
- 1Mbit
- 512kbps
- 320kbps
- 256kbps
- 128kbps
- 64kbps

We should inform the user that these are maximum bitrates, we obviously can't upgrade 320kbps to 5Mbit ;)

I look forward to your pull request! Let me know if you need any other pointers.

**Comment by hexadecimaldev** [comment]
It may take me some time due to other commitments and this will be my first time working with react-native and react-native player. 

I will add 92kpbs to your list, as that may be useful for people on lower mobile plans and it sounds a little better than 64kpbs.

The only additional usability feature I am considering is:

`TouchableOpacity onPress={someHandler}` on

` <Label numberOfLines={1} $overflow>`
        `{albumTrack.Codec?.isDirectPlay ? t('direct-play') : t('transcoded')}`
` </Label>`
 
To enable a real-time toggle back to direct-play, which would be most useful when users are back on either their local Wi-Fi or an unlimited Wi-Fi. 

I am having a think about the best way to implement that, that requires minimal changes to your code.

Finally, do you have a rough time-frame on when you will migrate to sqlite? Would it be best to wait for that?

**Comment by leinelissen** [comment]
Fair enough! The usability feature sounds great. The migration to SQLite is a work in progress and quite rough, and has no deadline. Just go ahead and implement against the current `main` and I will rebase your changes if necessary.

**Comment by jonhoo** [comment]
@hexadecimaldev Did you have a chance to make any progress on this? Would be such a nice quality of life addition!

Also, I wonder if we should give the option of MP3 V0? That arguably sits at the best trade-off space between quality and size.

**Comment by hexadecimaldev** [comment]
@jonhoo I did and I didn't. Requesting the desired bitrate from the server is easy, however I ran into issues with the queue. The queue is built all at once and you need to drop and rebuild the queue on bitrate change. The other issue is Apple devices also need a different container to support scrubbing when you request a transcode from the server, if you don't do that the seek doesn't work correctly.

Ultimately, the scope of these changes became too large for me to confidently test across all supported platforms, and I was constantly running into issues trying to build for my own iOS device. I've been in Wi-Fi range a lot more, so my requirement for a lower bitrate dropped off.

Oddly the lyrics feature doesn't work for me in the prod version but did in the dev builds that I ended up needing Xcode to compile with a self cert.
